mscore/preferenceslistwidget.cpp

```diff
@@ -131,187 +131,707 @@ void PreferenceItem::save(QVariant value)
 //   ColorPreferenceItem
 //---------------------------------------------------------
 
-ColorPreferenceItem::ColorPreferenceItem(QString name)
+ColorPreferenceItem::ColorPreferenceItem(QString name, std::function<void()> applyFunc, std::function<void()> updateFunc)
       : PreferenceItem(name),
         _initialValue(preferences.getColor(name)),
-        _editor(new Awl::ColorLabel)
+        _editorColorLabel(new Awl::ColorLabel)
       {
-      _editor->setColor(_initialValue);
-      _editor->setSizePolicy(QSizePolicy::Minimum, QSizePolicy::Fixed);
+      _editorColorLabel->setColor(_initialValue);
+      _editorColorLabel->setSizePolicy(QSizePolicy::Minimum, QSizePolicy::Fixed);
+      connect(_editorColorLabel, &Awl::ColorLabel::colorChanged, this, &PreferenceItem::editorValueModified);
+      applyFunction = applyFunc;
+      updateFunction = updateFunc;
+      }
+
+ColorPreferenceItem::ColorPreferenceItem(QString name, Awl::ColorLabel* editor, std::function<void()> applyFunc, std::function<void()> updateFunc)
+      : PreferenceItem(name),
+        _initialValue(preferences.getColor(name)),
+        _editorColorLabel(editor)
+      {
+      _editorColorLabel->setColor(_initialValue);
+      _editorColorLabel->setSizePolicy(QSizePolicy::Minimum, QSizePolicy::Fixed);
+      connect(_editorColorLabel, &Awl::ColorLabel::colorChanged, this, &PreferenceItem::editorValueModified);
+      applyFunction = applyFunc;
+      updateFunction = updateFunc;
```

Comment:
I think `applyFunction` should be renamed `_applyFunction`, `updateFunction` also
```
_applyFunction = applyFunc;
_updateFunction = updateFunc;
```